server/datastore/mysql/vpp.go

```diff
@@ -542,3 +542,46 @@ WHERE
 
 	return user, act, nil
 }
+
+func (ds *Datastore) GetVPPTokenByLocation(ctx context.Context, loc string) (*fleet.VPPToken, error) {
+	const stmt = `
+SELECT
+	vpt.id,
+	vpt.organization_name,
+	vpt.location,
+	vpt.renew_at,
+	vpt.token,
+	vpt.team_id,
+	vpt.null_team_type,
+	COALESCE(tm.name, CASE
+		WHEN vpt.null_team_type = 'noteam' THEN 'No team'
+		WHEN vpt.null_team_type = 'appteams' THEN 'All teams'
+		ELSE ''
+	END) as team_name
```

Comment:
For now I used this approach to populate the `"teams"` field that we want in the JSON payloads, but will need to integrate with Dante's PR when it lands to ensure we have a compatible approach in all "read" methods of the VPP tokens.